harness/determined/layers/_workload_manager.py

```diff
@@ -144,32 +143,42 @@ def _respond(in_response: workload.Response) -> None:
             metrics = in_response["metrics"]
             metrics = cast(workload.Metrics, metrics)
 
-            batch_metrics = metrics["batch_metrics"]
-
-            # Sanity-check training metrics.
-            det.util.validate_batch_metrics(batch_metrics)
-            check_len(batch_metrics, wkld.num_batches)
-
-            for callback in self.callbacks:
-                callback.on_train_step_end(
-                    wkld.step_id, wkld.num_batches, wkld.total_batches_processed, metrics
-                )
+            if in_response.get("invalid_hp", False):
+                out_response = {
+                    "type": "WORKLOAD_COMPLETED",
+                    "workload": wkld,
+                    "start_time": start_time,
+                    "end_time": _current_timestamp(),
+                    "metrics": metrics,
+                }
+                out_response["exited_reason"] = "INVALID_HP"
+                respond(out_response)
+            else:
```

Comment:
Is it valid to just `return` here, rather than add an extra layer of indentation to the whole rest of the function?

Reply:
Yep, good call here.